### PR TITLE
add yandex browser

### PR DESCRIPTION
**Pull request by 13werwolf13**
due to the fact that yandex-browser is finally out of beta.

### CODE REVIEW

**Comment by asdil12** [comment]
You have conflicting queries, so this won't work.
Also it would make more sense to put this into a single file (and class) and ask the user, which version they want to install.
Especially as the repo is the same for beta and stable.
See https://github.com/openSUSE/opi/blob/master/opi/plugins/ms_edge.py for an example how this can be done.

**Comment by 13werwolf13** [comment]
> You have conflicting queries, so this won't work. Also it would make more sense to put this into a single file (and class) and ask the user, which version they want to install. Especially as the repo is the same for beta and stable. See https://github.com/openSUSE/opi/blob/master/opi/plugins/ms_edge.py for an example how this can be done.

sorry but the repos are NOT the same. However, I understood what to do and will do.
However, I did not understand about the conflict?

**Comment by asdil12** [comment]
Oh yea - sorry the repo really seems to be different. But is should still work with a single class. I just don't want to clutter the list of plugins.

About the conflict:
```
queries = ('yandex-browser-stable', 'yandex-browser', 'yandex')
queries = ('yandex-browser-beta', 'yandex-browser', 'yandex')
```
... must not have entries that exist in any other plugin - how should opi otherwise know, which plugin it should run.
I updated the testsuite to check for this case.

**Comment by 13werwolf13** [comment]
> ... must not have entries that exist in any other plugin - how should opi otherwise know, which plugin it should run. I updated the testsuite to check for this case.

now I understand. well, I thought that he would give a choice as in a normal text search. i was wrong. I will fix it and send a commit today, but a little later. Thanks.

**Comment by 13werwolf13** [comment]
@asdil12 I'm not sure if I did everything right, but I hope so.

File: opi/plugins/yandex-browser.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+			'yandex-browser-stable',
+			'yandex-browser-beta',
+		])
+
+		release = option.split('-')[2]
+		print("You have chosen %d", release)
+
+		opi.add_repo(
+			filename = 'option',
+			name = 'option',
```
Comment (asdil12) [inline opi/plugins/yandex-browser.py:26]:
```suggestion
			filename = option,
			name = option,
```

File: opi/plugins/yandex-browser.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+			url = f'https://repo.yandex.ru/yandex-browser/rpm/{release}/$basearch/',
+			gpgkey = 'https://repo.yandex.ru/yandex-browser/YANDEX-BROWSER-KEY.GPG',
+			gpgcheck = False
+		)
+
+		opi.install_packages(['option'])
+		opi.ask_keep_repo('yandex-browser')
```
Comment (asdil12) [inline opi/plugins/yandex-browser.py:33]:
```suggestion
		opi.install_packages([option])
		opi.ask_keep_repo(option)
```

**Review by asdil12 — CHANGES_REQUESTED** [review]
See my comments - also please squash your changes into a single commit and test that everything works.

File: opi/plugins/yandex-browser.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import opi
+from opi.plugins import BasePlugin
+import subprocess
```
Comment (asdil12) [inline opi/plugins/yandex-browser.py:3]:
```suggestion
```

**Review by asdil12 — CHANGES_REQUESTED** [review]

**Comment by 13werwolf13** [comment]
@asdil12 sorry, i not know how do that
i do
```
git rebase -i fd4008d7fb53a031a28e25c82b9a023871bcbe84
git push --force
```
and i get
```
Everything up-to-date
```

or should I delete this PR and create a new one from new branch?

**Comment by 13werwolf13** [comment]
ah, I figured out how to do it .. fixed.

**Review by asdil12 — APPROVED** [review]

**Pull request merged**